### PR TITLE
Cross link API docs and concepts docs more throughly

### LINKED ISSUE #8079: Improve cross links between `Usage` and `API Documentation` sections in the docs

### PR DESCRIPTION
**Pull request by sydney-runkle**
## Change Summary

Cross link API docs and concepts docs to improve navigability of docs

## Related issue number

Closes #8079

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [x] Tests pass on CI
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by sydney-runkle** [comment]
Please review

**Review by samuelcolvin — APPROVED** [review]
Otherwise LGTM, this is awesome.

File: docs/concepts/fields.md [hunk 1]
```diff
@@ -195,8 +196,11 @@ print(user.model_dump(by_alias=True))  # (2)!
 
 
     user = User(name='johndoe')  # (1)!
+    user = User(username='johndoe')  # (2)!
     ```
-    1. Now VSCode will not show a warning
+
```
[the remark this answers is not in the document]
Comment (sydney-runkle) [reply docs/concepts/fields.md:201]: Yeah I thought so as well, but if we don't do 1, 2, 3, 4 the format doesn't work correctly.

[the remark this answers is not in the document]
Comment (sydney-runkle) [reply docs/concepts/fields.md:201]: Actually, if I fix the middle number, it works. Will push that change. So now it's numbered by each example as would be intuitive 👍

**Pull request merged**